discordbot/cogs/internet.py

```diff
@@ -39,30 +53,35 @@ def __init__(self, bot: DiscordBot):
             user = user.display_name
         else:
             user = ctx.message.author.display_name
-        await ctx.send("https://robohash.org/{}.png".format(user.replace(" ", "%20")))
+        k = "https://robohash.org/{}.png".format(user.replace(" ", "%20"))
+        o = wget.download(k)
+        await ctx.send(file=o)
 
     @commands.command()
-    async def xkcd(self, ctx, query=""):
-        """Queries a random XKCD comic.
-
-        Do `^xkcd <number from 1-1662>` to pick a specific comic."""
-        if not query:
-            i = randint(1, 1662)
-            url = "https://xkcd.com/{}/".format(i)
-        elif query.isdigit() and 1 <= int(query) <= 1796:
-            url = "https://xkcd.com/{}/".format(query)
-        elif int(query) <= 0 or int(query) >= 1796:
-            await ctx.send("It has to be between 1 and 1796!")
-        elif not query.isdigit():
-            await ctx.send("You have to put a number!")
-        else:
-            await ctx.send("I don't know how you managed to do it, but you broke it.")
-        with aiohttp.ClientSession() as session:
-            async with session.get(url) as resp:
-                r = await resp.read()
-        resp = BSoup(r, "html.parser")
-        await ctx.send(
-            ":mag:**" + resp("img")[1]["alt"] + "**\nhttp:" + resp("img")[1]["src"] + "\n" + resp("img")[1]["title"])
+    async def xkcd(ctx, num=None):
+        """Looks up a xkcd comic"""
+        a = requests.get('https://xkcd.com/{}/info.0.json'.format(num))
+        r = requests.get('https://xkcd.com/info.0.json')
+        if num == None and r.status_code == 200:
+            k = await bot.say('Getting your comic...')
+            r = requests.get('https://xkcd.com/info.0.json')
+            a = json.loads(r.text)
+            ra = a['month'] + '/' + a['day'] + '/' + a['year']
+            o = wget.download(a['img'])
+            await bot.edit_message(k, "**Date Released**: {}\n**Comic Number**: {}\n**Title**: {}\n**Image**: ".format(ra, a['num'], a['title']))
+            await bot.upload(o)
```

Comment:
When you submit the code, it should be at least compatible, seeing that you submitted the proper function for uploading on line 58